src/strings-in-block/class-compose.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace WPML\PB\Gutenberg\StringsInBlock;
+
+class Compose implements StringsInBlock {
+
+	/** @var StringsInBlock[] $parsers */
+	private $parsers = [];
+
+	/**
+	 * @param StringsInBlock[] $parsers
+	 */
+	public function __construct( array $parsers ) {
+		$this->parsers = $parsers;
+	}
+
+	/**
+	 * @param \WP_Block_Parser_Block $block
+	 *
+	 * @return array
+	 */
+	public function find( \WP_Block_Parser_Block $block ) {
+		$strings = [];
+
+		foreach ( $this->parsers as $parser ) {
+			$strings = array_merge( $strings, $parser->find( $block ) );
+		}
+
+		/**
+		 * String in block filter.
+		 *
+		 * When post with Gutenberg blocks is being send for translation
+		 * WPML parses blocks to find translatable strings. With this filter
+		 * you can register additional strings to be added to Translation
+		 * Package and then translated
+		 *
+		 * @param array                  $strings already found strings.
+		 * @param \WP_Block_Parser_Block $block block being parsed.
+		 */
+		return apply_filters( 'wpml_found_strings_in_block', $strings, $block );
```

Comment:
@strategio I'm not sure if this should be here. To me it seems like another thing that could implement `StringsInBlock`. 
Then the factory would change to:
```php
$string_parsers = [
			new WPML\PB\Gutenberg\StringsInBlock\HTML( $config_option ),
			new WPML\PB\Gutenberg\StringsInBlock\Attributes( $config_option ),
			new WPML\PB\Gutenberg\StringsInBlock\APIFilters(),
		];
```

You'll need to change the signature of the `find` function to also receive the strings.

This will then keep everything separate.

Reply:
@brucepearson, I understand your suggestion but it does not guarantee that the `APIFilters` parser will be at the end (and the order is not really testable in the factory).

The order is important because I have one case where we need to remove a string from the found collection (https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6611).